miraclelinux/meta-debian
# libsepol: Update from 2.8-1 to 2.8-1+deb10u1

# Purpose of pull request

This version fixes following CVEs.

- CVE-2021-36084
- CVE-2021-36085
- CVE-2021-36086
- CVE-2021-36087

ELA: ELA-1209-1

This version contains patches so removed DEBIAN_QUILT_PATCHES line.

# Test

1. Build package
2. Run image and check files

## Build libsepol

Set following environment values.

```
export TEST_PACKAGES="libsepol"
export TEST_DISTROS="deby"
export TEST_MACHINES="qemuarm64"
export COMPOSE_HTTP_TIMEOUT=7200
```

Run build_test

```
cd docker
make build_test
```
## Build core-image-minimal

Add following line in the conf/local.conf.

```
IMAGE_INSTALL_append = " libsepol ldd file"
```

## Check file

Start image

```
runqemu nographic qemuarm64
```

Check file.

```
ldd /lib/libsepol.so.1
file /lib/libsepol.so.1
```

# Test result

## Build libseopl

```
$ make build_test
docker-compose run --rm build_test
WARNING: The PTEST_RUNNER_TIMEOUT variable is not set. Defaulting to a blank string.
WARNING: The QEMU_PARAMS variable is not set. Defaulting to a blank string.
WARNING: The IMAGE_ROOTFS_EXTRA_SPACE variable is not set. Defaulting to a blank string.
WARNING: The http_proxy variable is not set. Defaulting to a blank string.
WARNING: The https_proxy variable is not set. Defaulting to a blank string.
WARNING: The no_proxy variable is not set. Defaulting to a blank string.
WARNING: The TEST_DISTRO_FEATURES variable is not set. Defaulting to a blank string.
WARNING: The TEST_ENABLE_SECURITY_UPDATE variable is not set. Defaulting to a blank string.
NOTE: Setup build directory.
You had no conf/local.conf file. This configuration file has therefore been
created for you with some default values. You may wish to edit it to, for
example, select a different MACHINE (target hardware). See conf/local.conf
for more information as common configuration options are commented.

You had no conf/bblayers.conf file. This configuration file has therefore been
created for you with some default values. To add additional metadata layers
into your configuration please add entries to conf/bblayers.conf.

The Yocto Project has extensive documentation about OE including a reference
manual which can be found at:
    http://yoctoproject.org/documentation

For more information about OpenEmbedded see their website:
    http://www.openembedded.org/

NOTE: Testing distro deby ...
NOTE: Testing machine qemuarm64 ...
NOTE: These recipes will be tested: libsepol
NOTE: Building libsepol ...
NOTE: Build libsepol: PASS
```

## File check result

```
# ldd /lib/libsepol.so.1 
        linux-vdso.so.1 (0x0000ffff88eb1000)
        libc.so.6 => /lib/libc.so.6 (0x0000ffff88c59000)
        /lib/ld-linux-aarch64.so.1 (0x0000ffff88e83000)
# file /lib/libsepol.so.1 
/lib/libsepol.so.1: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, BuildID[sha1]=38eefb3bde95d2511d1bc2b22d68d9d9e1e66185, stripped

```
